internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -929,6 +938,15 @@ func resourceCosmosDbAccountCreate(d *pluginsdk.ResourceData, meta interface{})
 		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
+	// NOTE: this is to work around the issue here: https://github.com/Azure/azure-rest-api-specs/issues/27596
+	// Once the above issue is resolved we shouldn't need this check and update anymore
+	if d.Get("create_mode").(string) == string(cosmosdb.CreateModeRestore) {
```

Comment:
It would be difficult to pull out into it's own PR because it requires `minimal_tls_version` to be implemented first, otherwise the check wouldn't have purpose. The check just re-sends the original payload so that the TLS version is set correctly on the backend. 

It's kind of a chicken/egg scenario, since without Minimal TLS, there's no action to be taken because the value won't be in the state.

So far as writing a test, I believe it is possible, but we would be checking for the presence of a bug. It would just be checking that the TLS version was first incorrectly set, then correctly set. This would work, but as soon as MS fixes it on their end, that test would break.